src/pymordemos/tr.py

```diff
@@ -0,0 +1,114 @@
+from pymor.algorithms.greedy import rb_greedy
+from pymor.algorithms.tr import coercive_rb_trust_region
+from pymor.parameters.functionals import MinThetaParameterFunctional
+from pymor.reductors.coercive import CoerciveRBReductor
+from pymordemos.linear_optimization import create_fom
+
+
+def main(
+    output_number: int = Argument(..., help='Selects type of output functional [0, 1], '
+                                         + 'where 0 stands for linear and 1 for a quadratic output'),
+    grid_intervals: int = Argument(..., help='Grid interval count.'),
+    training_samples: int = Argument(..., help='Number of samples used for training the reduced basis.')
+):
+    assert output_number in [0, 1]
+    if output_number == 0:
+        fom, mu_bar = create_fom(grid_intervals, output_type='l2')
+    else:
+        fom, mu_bar = create_fom(grid_intervals, output_type='quadratic')
+
+    parameter_space = fom.parameters.space(0, np.pi)
+    initial_guess = fom.parameters.parse([0.25, 2.5])
+
+    ####################
+    # FOM optimization #
+    ####################
+
+    tic = perf_counter()
+    reference_mu, reference_data = error_aware_bfgs(fom, parameter_space)
+    toc = perf_counter()
+    reference_data['time'] = toc - tic
+
+    #########################
+    # ROM optimization BFGS #
+    #########################
+
+    coercivity_estimator = MinThetaParameterFunctional(fom.operator.coefficients, mu_bar)
+
+    training_set = parameter_space.sample_uniformly(training_samples)
+
+    greedy_reductor = CoerciveRBReductor(fom, product=fom.energy_product, coercivity_estimator=coercivity_estimator)
+    greedy_data = rb_greedy(fom, greedy_reductor, training_set, atol=1e-4)
+    greedy_rom = greedy_data['rom']
+
+    tic = perf_counter()
+    bfgs_mu, bfgs_data = error_aware_bfgs(greedy_rom, parameter_space)
+    toc = perf_counter()
+    bfgs_data['time'] = toc - tic + greedy_data['time']
+
+    ################################
+    # ROM optimization adaptive TR #
+    ################################
+
+    reductor = CoerciveRBReductor(fom, product=fom.energy_product, coercivity_estimator=coercivity_estimator)
+
+    tic = perf_counter()
+    tr_mu, tr_data = coercive_rb_trust_region(reductor, parameter_space=parameter_space,
+                                              initial_guess=initial_guess)
+    toc = perf_counter()
+    tr_data['time'] = toc - tic
+
+    #############
+    # Reporting #
+    #############
+
+    reference_output = fom.output(reference_mu)[0, 0]
+    bfgs_output = fom.output(bfgs_mu)[0, 0]
+    tr_output = fom.output(tr_mu)[0, 0]
+
+    report(reference_mu, reference_output, reference_mu, reference_output, reference_data,
+           parameter_space.parameters.parse, descriptor=' of optimization with FOM model')
+    report(bfgs_mu, bfgs_output, reference_mu, reference_output, bfgs_data,
+           parameter_space.parameters.parse, descriptor=' of optimization with fixed ROM model and BFGS method')
```

Comment:
Would be nice to also see the dimensions of the ROM/number of FOM evaluations here.

Reply:
I added more information.